### PR TITLE
Cleanup global styles

### PR DESCRIPTION
**Pull request by pettinarip**
## Description

1. Removed all previous reset css rules in `GlobalStyles` in favor of the reset rules applied by Chakra.
2. Refactor: moved all the _still_ necessary global styles remaining inside `GlobalStyles` into the Chakra theme folder. To have everything in one place.
3. Deleted the `GlobalStyles` component.

## Related Issue

#6374

### CODE REVIEW

File: src/@chakra-ui/gatsby-plugin/semanticTokens.ts [hunk 1]
```diff
@@ -34,10 +34,10 @@ const semanticTokens = {
...
-    body: { _light: "grey.700", _dark: "grey.100" },
-    bodyLight: { _light: "grey.500", _dark: "grey.100" },
-    disabled: { _light: "grey.400", _dark: "grey.500" },
-    background: { _light: "white", _dark: "grey.700" },
+    body: { _light: "gray.700", _dark: "gray.100" },
+    bodyLight: { _light: "gray.500", _dark: "gray.100" },
+    disabled: { _light: "gray.400", _dark: "gray.500" },
+    background: { _light: "white", _dark: "gray.700" },
     success: "green.500",
     error: "red.600",
```
Comment (minimalsm) [inline src/@chakra-ui/gatsby-plugin/semanticTokens.ts:42]: This has been changed to red 500 on the design system (red 600 was a typo)

Comment (pettinarip) [reply src/@chakra-ui/gatsby-plugin/semanticTokens.ts:42]: Good catch, ty

File: src/@chakra-ui/gatsby-plugin/foundations/colors.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 export type Colors = typeof colors
 
 const colors = {
-  grey: {
+  gray: {
```
Comment (corwintines) [inline src/@chakra-ui/gatsby-plugin/foundations/colors.ts:4]: I see in the figma this is spelt `grey`. Should that be updated there to reflect this change. I have no strong preference on the spelling we use, just should be consistent.

File: src/components/Layout.tsx [hunk 3]
```diff
@@ -7,7 +7,6 @@ import { IntlProvider } from "react-intl"
 import { LocaleProvider } from "gatsby-theme-i18n"
 
 import { lightTheme, darkTheme } from "../theme"
-import GlobalStyle from "./GlobalStyle"
```
Comment (corwintines) [inline src/components/Layout.tsx:10]: This was the only use of this component, so I think we can delete it :)

File: src/@chakra-ui/gatsby-plugin/foundations/colors.ts (hunk 2 above)
Comment (pettinarip) [reply src/@chakra-ui/gatsby-plugin/foundations/colors.ts:4]: Good call, yes.

To clarify, Chakra is using `gray`. Calling @nloureiro to update those names in the design system and keep the consistency.

**Review by corwintines — APPROVED** [review]
LGTM 🚀

**Pull request merged**